waltid-libraries/waltid-crypto-ios/src/iosMain/kotlin/id/walt/crypto/RSA.ios.kt

```diff
@@ -0,0 +1,210 @@
+package id.walt.crypto
+
+import id.walt.crypto.keys.Key
+import id.walt.crypto.keys.KeyMeta
+import id.walt.crypto.keys.KeyType
+import id.walt.platform.utils.ios.DS_Operations
+import id.walt.platform.utils.ios.RSAKeyUtils
+import io.ktor.util.encodeBase64
+import kotlinx.cinterop.ptr
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.jsonObject
+import platform.CoreFoundation.CFDictionaryAddValue
+import platform.CoreFoundation.CFDictionaryCreateMutable
+import platform.CoreFoundation.CFMutableDictionaryRef
+import platform.CoreFoundation.kCFAllocatorDefault
+import platform.CoreFoundation.kCFBooleanTrue
+import platform.CoreFoundation.kCFTypeDictionaryKeyCallBacks
+import platform.CoreFoundation.kCFTypeDictionaryValueCallBacks
+import platform.Foundation.CFBridgingRelease
+import platform.Foundation.CFBridgingRetain
+import platform.Foundation.NSNumber
+import platform.Security.SecItemDelete
+import platform.Security.SecKeyCreateRandomKey
+import platform.Security.kSecAttrApplicationLabel
+import platform.Security.kSecAttrApplicationTag
+import platform.Security.kSecAttrIsPermanent
+import platform.Security.kSecAttrKeyClassPrivate
+import platform.Security.kSecAttrKeyClassPublic
+import platform.Security.kSecAttrKeySizeInBits
+import platform.Security.kSecAttrKeyType
+import platform.Security.kSecAttrKeyTypeRSA
+import platform.Security.kSecKeyAlgorithmRSASignatureDigestPKCS1v15SHA256
+
+class RSAKey private constructor(
```

Comment:
reason why I splitted to separate classes:
-  no umbrella generator. I can specify directly what key type I need and create it. 
-  there is much more code for iOS needed than for android, thats why I choose to separate to classes instead of having when statement in every method.